src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentchecker/DuplicateFilenameDocumentCheckerTest.java

```diff
@@ -0,0 +1,972 @@
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .bulkPrintCoverSheetIntv2(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .bulkPrintCoverSheetIntv3(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .bulkPrintCoverSheetIntv4(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .formA(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .bulkPrintCoverSheetAppConfidential(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .bulkPrintCoverSheetResConfidential(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .divorceUploadPetition(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .uploadMediatorDocument(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .uploadMediatorDocumentPaperCase(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .formC(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .formG(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .refusalOrderPreviewDocument(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .latestRefusalOrder(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .refusalOrderAdditionalDocument(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .latestDraftHearingOrder(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .orderApprovedCoverLetter(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .outOfFamilyCourtResolution(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .additionalListOfHearingDocuments(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .variationOrderDocument(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .consentVariationOrderDocument(DUPLICATED_CASE_DOCUMENT)
+                .build())
+        );
+    }
```

Comment:
Are these all used?

Reply:
Ah, just clocked that it's a parameterized test.  So guess they are all used?

Reply:
Basically the arguments are setting a CaseDocument with the target filename to all possible locations in `FinremCaseData` including the nested fields and the element of a List. 

Therefore the same testing result will be expected in the test:
`void testGetWarningsOnExistingCase(FinremCaseData caseData)`